bin/update_pythons.py

```diff
@@ -190,21 +190,22 @@ def __init__(self) -> None:
             # Removing the prefix, Python 3.9 would use: release["name"].removeprefix("Python ")
             version = Version(release["name"][7:])
 
-            if not version.is_prerelease and not version.is_devrelease:
-                uri = int(release["resource_uri"].rstrip("/").split("/")[-1])
-                self.versions_dict[version] = uri
+            uri = int(release["resource_uri"].rstrip("/").split("/")[-1])
+            self.versions_dict[version] = uri
 
     def update_version_macos(
         self, identifier: str, version: Version, spec: Specifier
     ) -> ConfigMacOS | None:
-        sorted_versions = sorted(v for v in self.versions_dict if spec.contains(v))
+
+        unsorted_versions = cast(Iterable[Version], spec.filter(self.versions_dict))
```

Comment:
This is really powerful; it gets the non-pre-releases unless it's all pre-releases; but had to mess with types a bit and manually cast, fixed in https://github.com/pypa/packaging/pull/430

Reply:
I've not taken the time to understand what's going on in there yet. All I can say is that it works !

Reply:
`spec.filter(x) is like `v for v in self.versions_dict if spec.contains(v)` _except_ that if _no_ versions match, it tries again with pre-releases enabled (just like pip would). Exactly what we want here. :)

Reply:
This is super handy, indeed! Perhaps worth a comment to that effect?

Reply:
@henryiii, could you do this one please ?

Reply:
Ahh, missed this. Thought I saw it in the email, then couldn't find it in the discussion. Now see it in the code review.